src/plugins/controls/public/control_group/editor/control_editor.tsx

```diff
@@ -112,49 +112,14 @@ export const ControlEditor = ({
     embeddable ? embeddable.getInput().fieldName : undefined
   );
 
-  const doubleLinkFields = (dataView: DataView) => {
-    // double link the parent-child relationship specifically for case-sensitivity support for options lists
-    const fieldRegistry: DataControlFieldRegistry = {};
-
-    for (const field of dataView.fields.getAll()) {
-      if (!fieldRegistry[field.name]) {
-        fieldRegistry[field.name] = { field, compatibleControlTypes: [] };
-      }
-      const parentFieldName = (field.subType as IFieldSubTypeMulti)?.multi?.parent;
-      if (parentFieldName) {
-        fieldRegistry[field.name].parentFieldName = parentFieldName;
-
-        const parentField = dataView.getFieldByName(parentFieldName);
-        if (!fieldRegistry[parentFieldName] && parentField) {
-          fieldRegistry[parentFieldName] = { field: parentField, compatibleControlTypes: [] };
-        }
-        fieldRegistry[parentFieldName].childFieldName = field.name;
-      }
-    }
-    return fieldRegistry;
-  };
-
-  const fieldRegistry = useMemo(() => {
-    if (!state.selectedDataView) return;
-    const newFieldRegistry: DataControlFieldRegistry = doubleLinkFields(state.selectedDataView);
-
-    const controlFactories = getControlTypes().map(
-      (controlType) => getControlFactory(controlType) as IEditableControlFactory
-    );
-    state.selectedDataView.fields.map((dataViewField) => {
-      for (const factory of controlFactories) {
-        if (factory.isFieldCompatible) {
-          factory.isFieldCompatible(newFieldRegistry[dataViewField.name]);
-        }
-      }
-
-      if (newFieldRegistry[dataViewField.name]?.compatibleControlTypes.length === 0) {
-        delete newFieldRegistry[dataViewField.name];
+  const [fieldRegistry, setFieldRegistry] = useState<DataControlFieldRegistry>();
+  useEffect(() => {
+    (async () => {
+      if (state.selectedDataView?.id) {
+        setFieldRegistry(await loadFieldRegistryFromDataViewId(state.selectedDataView.id));
       }
-    });
-
-    return newFieldRegistry;
-  }, [state.selectedDataView, getControlFactory, getControlTypes]);
+    })();
+  }, [state.selectedDataView]);
```

Comment:
Much cleaner to pull this out in to the `data_control_editor_tools` file, nice 🔥